packages/amazonq/test/unit/amazonq/apps/inline/completion.test.ts

```diff
@@ -0,0 +1,202 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import sinon from 'sinon'
+import { commands, languages } from 'vscode'
+import assert from 'assert'
+import { LanguageClient } from 'vscode-languageclient'
+import { AmazonQInlineCompletionItemProvider, InlineCompletionManager } from '../../../../../src/app/inline/completion'
+
+describe('InlineCompletionManager', () => {
+    let manager: InlineCompletionManager
+    let languageClient: LanguageClient
+    let sendNotificationStub: sinon.SinonStub
+    let registerProviderStub: sinon.SinonStub
+    let registerCommandStub: sinon.SinonStub
+    let executeCommandStub: sinon.SinonStub
+    let disposableStub: sinon.SinonStub
+    let sandbox: sinon.SinonSandbox
+    let getActiveSessionStub: sinon.SinonStub
+    let getActiveRecommendationStub: sinon.SinonStub
+
+    beforeEach(() => {
+        sandbox = sinon.createSandbox()
+
+        registerProviderStub = sandbox.stub(languages, 'registerInlineCompletionItemProvider')
+        registerCommandStub = sandbox.stub(commands, 'registerCommand')
+        executeCommandStub = sandbox.stub(commands, 'executeCommand')
+        sendNotificationStub = sandbox.stub()
+
+        // Create mock disposable
+        const mockDisposable = {
+            dispose: sandbox.stub(),
+        }
+        disposableStub = mockDisposable.dispose
+        registerProviderStub.returns(mockDisposable)
+
+        languageClient = {
+            sendNotification: sendNotificationStub,
+        } as unknown as LanguageClient
+
+        manager = new InlineCompletionManager(languageClient)
+        getActiveSessionStub = sandbox.stub(manager['sessionManager'], 'getActiveSession')
+        getActiveRecommendationStub = sandbox.stub(manager['sessionManager'], 'getActiveRecommendation')
+    })
+
+    afterEach(() => {
+        sandbox.restore()
+    })
+
+    describe('constructor', () => {
```

Comment:
do these constructor and disposable tests add any additional value? It looks like its just checking stubs are getting called

Reply:
agree they are not super necessary( Q wrote it for me ;) will remove)